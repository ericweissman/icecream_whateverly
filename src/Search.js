import React, { Component } from 'react';
import './css/Main.scss';

class Search extends Component {
  constructor() {
    super()
  }

  render() {
    return (
      <div className='search'>
        <i className="fas fa-search">
          <input type='text' placeholder="Search flavors" className='search-input' onChange={this.props.searchFor}></input>
        </i>
      </div>
    )
  }
}

export default Search;