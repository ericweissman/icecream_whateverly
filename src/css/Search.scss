@import '_variables.scss';

.search-input {
    text-align: center;
    margin-left: 7px;
    padding: 6px;
    border: 2px solid $skyblue-primary-color;
}

.fas {
    background-color: $skyblue-primary-color;
    border: 2px solid $skyblue-primary-color;
    border-top-left-radius: 15px;
    border-bottom-left-radius: 15px;
    font-size: 30px;
    color: #FFF;
    padding-left: 10px;
    border: 1px solid $grey;
}