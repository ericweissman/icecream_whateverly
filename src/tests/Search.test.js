import React from 'react';
import Search from '../Search';
import { shallow } from 'enzyme';

const searchForMock = jest.fn();

describe('Search', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Search searchFor={searchForMock} />
    )
  })

  it('should match snapshot when all data is passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call searchFor when input is changed', () => {
    wrapper.find('.search-input').simulate('change', { target: {value : ''}});
    expect(searchForMock).toBeCalled();
  });
})