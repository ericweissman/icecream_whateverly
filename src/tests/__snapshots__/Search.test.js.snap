// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot when all data is passed correctly 1`] = `
<div
  className="search"
>
  <i
    className="fas fa-search"
  >
    <input
      className="search-input"
      onChange={[MockFunction]}
      placeholder="search flavors"
      type="text"
    />
  </i>
</div>
`;
